tests/integration/household_composition/test_repeating_household.py

```diff
@@ -1,141 +1,76 @@
-            data={'action[start_questionnaire]': 'Start Questionnaire'},
-            follow_redirects=False)
-        self.first_page = resp.location
-
-    def tearDown(self):
-        super(TestRepeatingHousehold, self).tearDown()
+        self.launchSurvey('test', 'repeating_household')
+        self.post(action='start_questionnaire')
+        self.household_composition_url = self.last_url
 
     def test_change_repeat_answer_removes_all_answers_for_repeated_groups(self):
         # Given I add some people
-        household_composition_page = self.first_page
-
         form_data = MultiDict()
-        form_data.add("household-0-first-name", 'Joe')
-        form_data.add("household-0-middle-names", '')
-        form_data.add("household-0-last-name", 'Bloggs')
-        form_data.add("household-1-first-name", 'Jane')
-        form_data.add("household-1-middle-names", '')
-        form_data.add("household-1-last-name", 'Doe')
-        form_data.add("action[save_continue]", '')
-
-        resp = self.get_and_post_with_csrf_token(household_composition_page, data=form_data, follow_redirects=False)
-        person1_age_page = resp.location
+        form_data.add('household-0-first-name', 'Joe')
+        form_data.add('household-0-middle-names', '')
+        form_data.add('household-0-last-name', 'Bloggs')
+        form_data.add('household-1-first-name', 'Jane')
+        form_data.add('household-1-middle-names', '')
+        form_data.add('household-1-last-name', 'Doe')
+        self.post(form_data)
 
         # And provide details
-        navigate_to_page(self.client, person1_age_page)
-        form_data = MultiDict()
-        form_data.add("what-is-your-age", '9990')
-        form_data.add("action[save_continue]", '')
-        resp = self.get_and_post_with_csrf_token(person1_age_page, data=form_data, follow_redirects=False)
-        person1_shoe_size_page = resp.location
-
-        navigate_to_page(self.client, person1_shoe_size_page)
-        form_data = MultiDict()
-        form_data.add("what-is-your-shoe-size", '9991')
-        form_data.add("action[save_continue]", '')
-        resp = self.get_and_post_with_csrf_token(person1_shoe_size_page, data=form_data, follow_redirects=False)
-        person2_age_page = resp.location
-
-        navigate_to_page(self.client, person2_age_page)
-        form_data = MultiDict()
-        form_data.add("what-is-your-age", '9992')
-        form_data.add("action[save_continue]", '')
-        resp = self.get_and_post_with_csrf_token(person2_age_page, data=form_data, follow_redirects=False)
-        person2_shoe_size_page = resp.location
-
-        navigate_to_page(self.client, person2_shoe_size_page)
-        form_data = MultiDict()
-        form_data.add("what-is-your-shoe-size", '9993')
-        form_data.add("action[save_continue]", '')
-        self.get_and_post_with_csrf_token(person2_shoe_size_page, data=form_data, follow_redirects=False)
+        self.post({'what-is-your-age': '9990'})
```

Comment:
I guess Flask doesn't help; it should be simpler out the box to write clean integration tests; the docs are also not particularly helpful. Once you know enough to write tests it's already too late!